internal/controller/azurevalidator_controller.go

```diff
@@ -65,77 +61,51 @@ func (r *AzureValidatorReconciler) Reconcile(ctx context.Context, req ctrl.Reque
-	// its status, save it in k8s. Its status is a "state" and "conditions". We
-	// use "InProgress" as the state because this is a brand new validation
-	// result object and we haven't done anything meaningful yet. We use an
-	// an empty set of conditions for the same reason.
-	vr := &v8or.ValidationResult{}
+	// Get the active validator's validation result
+	vr := &vapi.ValidationResult{}
 	nn := ktypes.NamespacedName{
 		Name:      fmt.Sprintf("validator-plugin-azure-%s", validator.Name),
 		Namespace: req.Namespace,
 	}
 	if err := r.Get(ctx, nn, vr); err == nil {
-		res, err := v8ores.HandleExistingValidationResult(nn, vr, r.Log)
+		res, err := vres.HandleExistingValidationResult(nn, vr, r.Log)
 		if res != nil {
 			return *res, err
 		}
 	} else {
 		if !apierrs.IsNotFound(err) {
 			r.Log.V(0).Error(err, "unexpected error getting ValidationResult", "name", nn.Name, "namespace", nn.Namespace)
 		}
-		res, err := v8ores.HandleNewValidationResult(r.Client, constants.PluginCode, nn, vr, r.Log)
+		res, err := vres.HandleNewValidationResult(r.Client, constants.PluginCode, nn, vr, r.Log)
 		if res != nil {
 			return *res, err
 		}
 	}
 
-	// The above was just for checking the state of the validation result in
-	// k8s. It wasn't actually doing the validation. Now, we need to do the
-	// validation, which means to use Go code to interact with Azure via its Go
-	// SDK and then save the result in the validation result. It will be either
-	// a fail or a pass and the validators framework can read it and do stuff
-	// with it later.
-	//
-	// For now, this is just no op validation. It's a success result.
-
-	// Helper to make the new validation result object. Modeled after helpers
-	// that I saw in the AWS version. Later, this should be moved to somewhere
-	// else in this codebase.
-	buildValidatorResult := func() *types.ValidationResult {
-		// Pick a validation state
-		state := v8or.ValidationSucceeded
-
-		// Make a validation condition
-		latestCondition := v8or.DefaultValidationCondition()
-		latestCondition.Message = "This is just a test. We're pretending it passed."
-		latestCondition.ValidationRule = fmt.Sprintf("%s-%s", v8orconstants.ValidationRulePrefix, "validator-testrule")
-		latestCondition.ValidationType = constants.ValidationTypeTest
-
-		// Once both are built, put them into a validation result
-		return &types.ValidationResult{
-			Condition: &latestCondition,
-			State:     &state,
+	failed := &types.MonotonicBool{}
+
+	// Role Assignment rules
+	for _, rule := range validator.Spec.RoleAssignmentRules {
+		// Service is created once per rule because in order to create the
+		// service, it needs an Azure API client. And in order to create the
+		// the client, you need to know the subscription that will be queried.
+		// And you won't know which subscription that is until you read it from
+		// a rule spec.
+		client, err := azure_utils.NewRoleAssignmentsClient(rule.GetSubscriptionID())
```

Comment:
`rule.SubscriptionID`

Reply:
Need to keep this I think because subscription ID is part of the rules for Azure role assignments, and I'll need this to be testable too. I'm using an interface in this method and I think I will continue to need to use an interface.

I will check things over for opportunities to remove interfaces after I add a suite of unit tests.